PYTHON_CDM_Conversion_code/hourly_qff_to_cdm_obs_v1.py

```diff
@@ -1,34 +1,112 @@
 #!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
-Created on Thu Nov 11 16:31:58 2021
+Convert QFF files to CDM obs .psv files (one per station).
+
+CDM obs files have all variables, one after another.
+
+Call in one of three ways using:
+
+>python hourly_qff_to_cdm_obs_v1.py --station STATIONID
+>python hourly_qff_to_cdm_obs_v1.py --subset FILENAME
+>python hourly_qff_to_cdm_obs_v1.py --run_all
+>python hourly_qff_to_cdm_obs_v1.py --help
+#Created on Thu Nov 11 16:31:58 2021
 
 @author: snoone
+
+edited: snoone, February 2022
 """
 
 import os
 import glob
 import pandas as pd
 pd.options.mode.chained_assignment = None  # default='warn'
+import utils
+
+# Set the file extension for the subdaily psv files
+EXTENSION = 'qff'
+
+def main(station="", subset="", run_all=False, clobber=False):
+    """
+    Run processing of hourly QFF to CDM obs
+
+    Parameters
+    ----------
+
+    station : `str` 
+        Single station ID to process
+
+    subset : `str`
+        Path to file containing subset of IDs to process
+
+    run_all : `bool`
+        Run all files in the directory defined in the config file
+
+    clobber : `bool`
+        Overwrite existing files if they exist.  If False, will skip existing ones
+    """
+    # Read in either single file, list of files or run all
+
+    # Check for sensible inputs
+    if station != "" and subset != "" and all:
+        print("Please select either single station, list of stations run or to run all")
+        return
+    elif station == "" and subset == "" and not all:
+        print("Please select either single station, list of stations run or to run all")
+        return

```

Comment:
Note for us both - will need to check that this `if/elif` statement set covers all possible bad calls to this script (and others)